x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/action_connector_form/connector_form.tsx

```diff
@@ -169,6 +214,45 @@ const ConnectorFormComponent: React.FC<Props> = ({
     }
   }, [isFormModified, onFormModifiedChange]);
 
+  useEffect(() => {
```

Comment:
Connector type-specific logic should not be part of the connector_form component(even though that happens for the serializer, which will eventually be refactored in the future).

This `useEffect` should be moved to where the header components live, I think `stack_connectors/common/auth`